test-suite/generated-src/cpp/access_flags.hpp

```diff
@@ -9,16 +9,16 @@ namespace testsuite {
 
 enum class access_flags : unsigned {
     NOBODY = 0,
-    OWNER_READ = 1 << 0,
-    OWNER_WRITE = 1 << 1,
-    OWNER_EXECUTE = 1 << 2,
-    GROUP_READ = 1 << 3,
-    GROUP_WRITE = 1 << 4,
-    GROUP_EXECUTE = 1 << 5,
-    SYSTEM_READ = 1 << 6,
-    SYSTEM_WRITE = 1 << 7,
-    SYSTEM_EXECUTE = 1 << 8,
-    EVERYBODY = (1 << 9) - 1,
+    OWNER_READ = 1u << 0,
+    OWNER_WRITE = 1u << 1,
+    OWNER_EXECUTE = 1u << 2,
+    GROUP_READ = 1u << 3,
+    GROUP_WRITE = 1u << 4,
+    GROUP_EXECUTE = 1u << 5,
+    SYSTEM_READ = 1u << 6,
+    SYSTEM_WRITE = 1u << 7,
+    SYSTEM_EXECUTE = 1u << 8,
+    EVERYBODY = (1u << 9) - 1,
```

Comment:
I think we'd need to fix this as well to make the last bit truly usable.

Reply:
I mean the EVERYBODY case should be `|` of all members rather than `(1u << N) - 1`, which will overflow if N is 32.

Reply:
yes, I was expected it to be the same as is done for [cross-platform-djinni](https://github.com/cross-language-cpp/djinni-generator/releases/tag/v1.3.2)
Btw, just updated the logic of 'all' flag based on bitwise operation instead of using subtraction

Reply:
Awesome! Thank you!

Reply:
Could you check in all the regenerated files including objc/ts/wasm files?

Reply:
There's a CI check that verifies  all generated files to be up to date.

Reply:
I've just updated the generated enum files(for which the tests were failing)
there is a problem related to ts, as for ts there is no 'u' support
Seems we need to have a different enum generation for ts w/o 'u'

Reply:
I wonder if we can simply change the base type of the enum to int32_t.  I tested that this passes the compilation and the bit flags are correct.

Reply:
and since it uses signed value, it's available across all platforms that djinni needs to support (including java and js where unsigned is n/a).

Reply:
by changing it to int32 you reserve the first bit for sign, which will decrease the capacity if enum: i.e. 31 instead of 32

Reply:
We do need the all flags fix, otherwise 1<<32 will not compile

Reply:
the tricky point is related to ts, maybe we can have a separate logic of enum generation for it?

Reply:
no the first bit is interpreted as sign only when you view it as an integer,  but if you use it as bit flags (checking/comparing with bit operations) then it does not matter.

Reply:
```
#include <iostream>
#include <bitset>

enum class MyFlags1: int32_t {
    NO_FLAGS = 0,
    FLAG1 = 1 << 30,
    FLAG2 = 1 << 31,
    ALL_FLAGS = 0 | FLAG1 | FLAG2,
};

int main (int argc, char *argv[])
{
    auto a = MyFlags1::FLAG2;
    std::cout << std::bitset<32>(static_cast<int32_t>(a)) << std::endl;
    std::cout << (a == MyFlags1::FLAG2) << std::endl;
}
```
output: 
```
10000000000000000000000000000000
1
```

Reply:
and ALL_FLAGS is 11000000000000000000000000000000
which is also correct.

Reply:
It fails the compilation in the original code when the base type is unsigned only because when the sign bit is 1, the number value becomes negative, which cannot be casted to unsigned.

Reply:
yes the reason is casting. just conceptually use of signed bit is a annoying and if I was the owner of codebase, I'd avoid to have it.

btw, i32_t from functional point of view is also ok. So, as a 'good to go' - use int32_t and update 'ALL' prop's calculation(by using bitwise or instead of -)